pkg/util/oom/oom_linux.go

```diff
@@ -78,12 +75,22 @@ func applyOOMScoreAdj(pid int, oomScoreAdj int) error {
 	value := strconv.Itoa(oomScoreAdj)
 	var err error
 	for i := 0; i < maxTries; i++ {
-		if err = ioutil.WriteFile(oomScoreAdjPath, []byte(value), 0700); err != nil {
+		f, err := os.Open(oomScoreAdjPath)
+		if err != nil {
+			if syscallNotExists(err) {
+				return os.ErrNotExist
+			}
+			err = fmt.Errorf("failed to apply oom-score-adj to pid %d (%v)", pid, err)
+			continue
+		}
+		if _, err := f.Write([]byte(value)); err != nil {
 			if syscallNotExists(err) {
```

Comment:
Catchin ESRCH requires string parsing. Given that we will most likely retry if the write fails, which will trigger Open(..) subsequently, I don't think we need to worry about write errors. 
There is still a chance that we might not have enough retries, but the likelihood is less and I'd avoid spending more brain power on this.